src/OrbitCaptureClient/ApiEventProcessorTest.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <vector>
+
+#include "OrbitBase/Profiling.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "OrbitCaptureClient/ApiEventProcessor.h"
+#include "OrbitCaptureClient/CaptureEventProcessor.h"
+#include "OrbitCaptureClient/CaptureListener.h"
+#include "capture.pb.h"
+
+using orbit_client_protos::TimerInfo;
+
+using orbit_grpc_protos::ApiEvent;
+using orbit_grpc_protos::ClientCaptureEvent;
+
+namespace {
+
+// Empty implementation of CaptureListener used as helper to reduce the complexity in subclasses.
+class EmptyCaptureListener : public CaptureListener {
+ public:
+  void OnCaptureStarted(ProcessData&&,
+                        absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionInfo>,
+                        TracepointInfoSet, absl::flat_hash_set<uint64_t>) override {}
+  void OnTimer(const orbit_client_protos::TimerInfo&) override {}
+  void OnKeyAndString(uint64_t, std::string) override {}
+  void OnUniqueCallStack(CallStack) override {}
+  void OnCallstackEvent(orbit_client_protos::CallstackEvent) override {}
+  void OnThreadName(int32_t, std::string) override {}
+  void OnThreadStateSlice(orbit_client_protos::ThreadStateSliceInfo) override {}
+  void OnAddressInfo(orbit_client_protos::LinuxAddressInfo) override {}
+  void OnUniqueTracepointInfo(uint64_t, orbit_grpc_protos::TracepointInfo) override {}
+  void OnTracepointEvent(orbit_client_protos::TracepointEventInfo) override {}
+};
+
+// Test CaptureListener used to validate TimerInfo data produced by api events.
+class ApiEventCaptureListener : public EmptyCaptureListener {
+ public:
+  void OnTimer(const orbit_client_protos::TimerInfo& timer_info) override {
+    timers_.emplace_back(timer_info);
+  }
+  std::vector<TimerInfo> timers_;
+};
+
+// ApiTester exposes methods that mocks the Orbit API and internally creates grpc events that it
+// forwards to both a CaptureEventProcessor and a ApiEventProcessor. The test makes sure that both
+// processors forward the same information to their listener. Note that a CaptureEventProcessor owns
+// a ApiEventProcessor to which it forwards ApiEvent events. To help readability of test code, the
+// api methods return a ApiTester reference so that we can chain function calls.
```

Comment:
```suggestion
// ApiTester exposes methods that mocks the Orbit API and internally creates grpc events that it
// forwards to both a CaptureEventProcessor and an ApiEventProcessor. The test makes sure that both
// processors forward the same information to their listener. Note that a CaptureEventProcessor owns
// an ApiEventProcessor to which it forwards ApiEvent events. To help readability of test code, the
// api methods return an ApiTester reference so that we can chain function calls.
```

Or maybe this was on purpose. Just pointing it out.

Reply:
yep thanks